clients/venice-thin-client/src/main/java/com/linkedin/venice/client/store/AvroGenericReadComputeStoreClient.java

```diff
@@ -16,10 +16,28 @@
  * @param <V>
  */
 public interface AvroGenericReadComputeStoreClient<K, V> extends AvroGenericStoreClient<K, V> {
-  ComputeRequestBuilder<K> compute(
+  boolean isProjectionFieldValidationEnabled();
+
+  @Override
+  default ComputeRequestBuilder<K> compute() throws VeniceClientException {
+    return compute(Optional.empty(), Optional.empty(), 0);
+  }
+
+  default ComputeRequestBuilder<K> compute(
       Optional<ClientStats> stats,
       Optional<ClientStats> streamingStats,
-      long preRequestTimeInNS) throws VeniceClientException;
+      long preRequestTimeInNS) throws VeniceClientException {
+    return compute(stats, streamingStats, this, preRequestTimeInNS);
+  }
+
+  default ComputeRequestBuilder<K> compute(
+      Optional<ClientStats> stats,
+      Optional<ClientStats> streamingStats,
+      AvroGenericReadComputeStoreClient computeStoreClient,
+      long preRequestTimeInNS) throws VeniceClientException {
+    return new AvroComputeRequestBuilderV3<K>(computeStoreClient, getLatestValueSchema()).setStats(streamingStats)
```

Comment:
We have marked it deprecated because we don't want users to depend on that API of the client. For internal purposes, it is okay to use. The reason the function doesn't guarantee the actual schema being returned is that the thin-client refreshes schemas only on startup and when it sees a previously unseen schema. This is a known limitation of thin-client, but should not be a concern for fast-client, though as the metadata is periodically refreshed

Reply:
Thanks for the explanation. Resolving this.